testtools/iothub_test/src/iothubtest.c

```diff
@@ -2008,3 +2010,23 @@ IOTHUB_TEST_CLIENT_RESULT IoTHubTest_SendMessage(IOTHUB_TEST_HANDLE devhubHandle
 
     return result;
 }
+
+const char* IoTHubTest_GetTargetNetworkInterface()
+{
+    const char* result = NULL;
+    static char networkInterface[32];
```

Comment:
Actually why does this need to be a static char in 1st place?  Can't we just return value from getenv() or the DEFAULT_TARGET_NETWORK_INTERFACE to the caller without strcpy().

Or if we need this, I'd prefer caller to have to pass the buffer and length in.  I get the odds of this on being used on multiple threads seem unlikely but if that ever happens (since the function otherwise looks thread safe, but it's not with a static) it could cause issues.

Reply:
The reason I didn't do that was to keep the same code design used for the other functions that ready env vars. 
I wanted to keep this simple, but if I rather break with the design I'll just return the value read from getenv

Reply:
Interesting!  I hadn't looked at rest of file and just looking now.

So `IoTHubAccount_GetEventhubListenName` looks like we did this since there's some "post-processing" (which is fancier than saying sprint:)) and relying on nature of tests being single threaded.

But `IoTHubAccount_GetEventhubConsumerGroup` which the new function is closest to, also could just return the getenv result directly.

I'd personally vote that we return the getenv result (or the constant when env not set) directly from this function (and potentially change `IoTHubAccount_GetEventhubConsumerGroup` but I feel less strongly as that's legacy test code). 
 And otherwise leave other stuff alone.

Reply:
+1 for John

Reply:
> Can't we just return value from getenv() or the DEFAULT_TARGET_NETWORK_INTERFACE to the caller without strcpy().

According to the manual anything returned from `getenv()` must be copied or a future call to `getenv()` might change the previous return value.

> The implementation of getenv() is not required to be reentrant. The string pointed to by the return value of getenv() may be statically allocated, and ~~can be modified by a subsequent call to getenv()__, [putenv]